lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -59,15 +60,54 @@ def build_cmd(src)
           cmd
         end
 
-        def cleanup_files
-          src_file = "#{self.file_name}.c"
-          exe_file = "#{self.file_name}.exe"
 
+        def compile_cpp(src)
+          cmd = build_cpp_cmd(src)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def compile_cpp_files(files)
+          cmd = build_cpp_files_cmd(files)
+          print("#{cmd}\n") if show_compile_cmd
+          stdin_err, status = Open3.capture2e(cmd)
+          stdin_err
+        end
+
+        def build_cpp_cmd(src)
+          src_file = Tempfile.create(['scexec', ".cpp"])
+          File.write(src_file.path, src)
+          build_cpp_files(src_file.path)
+        end
+
+        def build_cpp_files_cmd(file_array)
+          cmd = [ mingw_bin ]
+          cmd << file_array.join(" ")
+          cmd << "-I #{INCLUDE_DIR}"
+          cmd << "-o #{outfile}"
```

Comment:
Would it not also make sense to change this for `build_cmd` to ensure consistency? Otherwise it may be a bit confusing to end users.

Reply:
Same scenario as before. I want to force fewer compiler and linker flags on the user than the C version of the method, but didn't want to change the original because of existing modules that use the mingw C compiler. Using the same flags in the `#build_cpp_files_cmd` method will break certain kinds of builds, for example, if a user wants to compile a `*.def` file along with a `*.cpp`  to define exports/forwards.